pyout/tabular.py

```diff
@@ -388,15 +402,27 @@ def __call__(self, row, style=None):
             via `columns`.
 
             Instead of a plain value, a column's value can be a tuple
-            of the form (initial_value, callable).  The callable will
-            be called asynchronously with no arguments and should
-            return the value with which to replace `initial_value`.
-
-            Using callable values requires some additional steps.  The
-            `ids` property should be set unless the first column
-            happens to be a suitable id.  The instance should also be
-            used as a context manager so that the program waits at the
-            end of the block for the return values.
+            of the form (initial_value, producer).  If a producer is
+            is a generator function or generator object, each item
+            procuced replaces `initial_value`.  Otherwise, a producer
+            should be a function that will be called with no arguments
+            and that returns the value with which to replace
+            `initial_value`.  For both generators and normal
+            functions, the execution will happen asynchronously.
+
+            The producer can return an update for multiple columns.
+            To do so, the keys of `row` should include a tuple with
+            the column names and the produced value should be a tuple
+            with the same order as the key or a mapping from column
+            name to the updated value.
+
+            Using the (initial_value, producer) form requires some
+            additional steps.  The `ids` property should be set unless
+            the first column happens to be a suitable id.  The
+            instance should also be used as a context manager so that
+            the program waits at the end of the block for the return
+            values.
```

Comment:
Can't it just just have some .wait() or alike which would wait until all callables are done and generators exhausted? I thought that context manager is just a sugaring on top of procedural implementation (sorry, hard to check on the phone ATM)

Reply:
> I thought that context manager is just a sugaring on top of procedural implementation

It is as of 256aaac (RF: Move Tabular.__exit__ code into wait method,
2018-01-05).  I'll update these docs to make that clearer.  Thanks.